### PR TITLE
Org Flyer Cell View 

### PR DESCRIPTION
**Pull request by cindy-x-liang**
## Overview

Added org flyer cell view for orgs to see the flyers they have posted. 

## Screenshots (delete if not applicable)

<details>

image

</details>

### CODE REVIEW

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            VStack(alignment: .leading, spacing: 8) {
+                organizationName
+                flyerTitle
+                flyerDate
+                flyerLocation
+            }
+        }
+        .padding(.bottom, 16)
+        
+    }
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:42]: Insert line here

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                Image(uiImage: urlImageModel.image ?? UIImage())
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+            } else {
+                SkeletonView()
+                    .shimmer(.mediumShimmer())
+            }
+        }
+        .frame(width: 80, height: 80)
+    }
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:57]: Insert line after

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    .fill().foregroundColor(Color.gray)
+                    .frame(width: 3, height: 3)
+                Circle()
+                    .fill().foregroundColor(Color.gray)
+                    .frame(width: 3, height: 3)
+            }
+
+        }
+        .padding(.bottom, -Constants.verticalSpacing)
+    }
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:79]: Insert line here

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    private var flyerTitle: some View {
+        Text(flyer.title)
+            .font(.newYorkMedium(size: 16))
+            .lineLimit(1)
+    }
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:84]: Insert line after

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        .frame(width: 80, height: 80)
+    }
+    private var organizationName: some View {
+        HStack(alignment: .top) {
+            Text(flyer.organization.name)
+                .font(.newYorkMedium(size: 10))
+                .lineLimit(2)
+
+            Spacer()
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:64]: Insert line after

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            HStack(alignment: .center, spacing: 2) {
+                Circle()
+                    .fill().foregroundColor(Color.gray)
+                    .frame(width: 3, height: 3)
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:68]: May or may or not be good to factor this number out into a constant

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    private var flyerLocation: some View {
+        HStack {
+            Image.volume.location
+                .foregroundColor(Color.black)
+
+            Text(flyer.location)
+                .font(.helveticaRegular(size: 12))
+                .lineLimit(1)
+        }
+    }
```
Comment (tiffany-pann) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:110]: Insert line after

**Review by tiffany-pann — APPROVED** [review]
Nice job on your first (iOS) PR! The screenshot was defo helpful :] Just some small nit-picky changes, no need to rerequest though 🙌🏻

File: Volume.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -446,8 +448,9 @@
 			isa = PBXGroup;
 			children = (
 				2E9349A92AC718C70098CAC8 /* FlyerUploadView.swift */,
+				EEC1C03B2AF6D1E800EDDCCC /* OrgFlyerCellView.swift */,
 				2E6C36102AF5E355004A54BB /* OrgsAdminView.swift */,
-				2EC811EB2AC6875500ED1ACC /* OrgsLoginView.swift */,
+                2EC811EB2AC6875500ED1ACC /* OrgsLoginView.swift */,
```
Comment (vinnie4k) [inline Volume.xcodeproj/project.pbxproj:453]: Can you indent this?

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+//  OrgFlyerCellView.swift
+//  Volume
+//
+//  Created by Cindy Liang on 11/4/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+
+struct OrgFlyerCellView: View {
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:11]: New line below.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    // MARK: - Properties
+
+    let flyer: Flyer
+    let navigationSource: NavigationSource
+
+    @StateObject var urlImageModel: URLImageModel
+    @EnvironmentObject private var userData: UserData
+    @ObservedObject var viewModel: FlyersView.ViewModel
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:19]: We don't need `viewModel` anymore.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    
+    // MARK: - Constants
+
+    private struct Constants {
+        static let horizontalSpacing: CGFloat = 8
+        static let verticalSpacing: CGFloat = 8
+    }
+    
+    var body: some View {
+
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:29]: Delete this line.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+            VStack(alignment: .leading, spacing: 8) {
+                organizationName
+                flyerTitle
+                flyerDate
+                flyerLocation
+            }
+        }
+        .padding(.bottom, 16)
+
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:41]: Delete this line.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+    private var organizationName: some View {
+        HStack(alignment: .top) {
+            Text(flyer.organization.name)
+                .font(.newYorkMedium(size: 10))
+                .lineLimit(2)
+
+            Spacer()
+            HStack(alignment: .center, spacing: 2) {
+                Circle()
+                    .fill().foregroundColor(Color.gray)
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:67]: Create a new line for `.foregroundColor(Color.gray)`

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift (hunk 6 above)
Comment (vinnie4k) [reply Volume/Views/Orgs Admin/OrgFlyerCellView.swift:68]: @cindy-x-liang I would filter this out and call it `circleSize: CGFloat = 3`

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+        .frame(width: 80, height: 80)
+    }
+    private var organizationName: some View {
+        HStack(alignment: .top) {
+            Text(flyer.organization.name)
+                .font(.newYorkMedium(size: 10))
+                .lineLimit(2)
+
+            Spacer()
+            HStack(alignment: .center, spacing: 2) {
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:65]: I would filter this HStack into the following:
```swift
private var tripleDotsButton: some View {
    // HStack goes here
}
```

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+                Circle()
+                    .fill().foregroundColor(Color.gray)
+                    .frame(width: 3, height: 3)
+                Circle()
+                    .fill().foregroundColor(Color.gray)
+                    .frame(width: 3, height: 3)
+                Circle()
+                    .fill().foregroundColor(Color.gray)
+                    .frame(width: 3, height: 3)
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:74]:
```swift
ForEach(0..<3) { _ in
    Circle()
        .fill()
        .foregroundColor(Color.gray)
        .frame(width: Constants.circleSize, height: Constants.circleSize)
}
```
I think this works.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+            }
+
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:76]: Remove this line.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+            Text(flyer.location)
+                .font(.helveticaRegular(size: 12))
+                .lineLimit(1)
+        }
+    }
+}
+
+// MARK: - Uncomment below if needed
+
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:114]: Delete this line.

**Review by vinnie4k — CHANGES_REQUESTED** [review]
Great work on your first iOS PR 🎉!! Please address my comments then request me again.

File: Volume/Views/Orgs Admin/OrgFlyerCellView.swift [hunk 18]
```diff
@@ -0,0 +1,122 @@
+//
+//  OrgFlyerCellView.swift
+//  Volume
+//
+//  Created by Cindy Liang on 11/4/23.
```
Comment (vinnie4k) [inline Volume/Views/Orgs Admin/OrgFlyerCellView.swift:5]: Nice :)

**Review by vinnie4k — APPROVED** [review]
Good to go. Great work!

**Pull request merged**